include/range/v3/detail/config.hpp

```diff
@@ -757,4 +757,10 @@ namespace ranges
 #endif
 #endif
 
+#if __has_include(<span>) && defined(__cpp_lib_span) && __cpp_lib_span > 0
```

Comment:
`__has_include` is a C++17 feature. Range-v3 supports C++14. Also, where does `__cpp_lib_span` get defined?

Reply:
Ah my bad. I see it's being used in other parts of the file, but conditional on its existence. Would this work?
```c++
#if defined(__has_include) && __has_include(<span>) && defined(__cpp_lib_span) && __cpp_lib_span > 0
#define RANGES_CXX_SPAN 1
#else
#define RANGES_CXX_SPAN 0
#endif
```

To the other question, `__cpp_lib_span` [should](https://en.cppreference.com/w/cpp/utility/feature_test) be defined when `<version>` is included: 
https://github.com/ericniebler/range-v3/blob/ffa477cb30e964d8578cb77ac8bc45f9743a94b0/include/range/v3/detail/config.hpp#L17-L24